app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -6,11 +6,37 @@ def create_destination
   def determine_placement
     availability_zone = placement
     options[:dest_availability_zone] = [availability_zone.try(:id), availability_zone.try(:name)]
-    signal :prepare_provision
+    signal :prepare_volumes
+  end
+
+  def prepare_volumes
+    # volumes example: [{name: "vol_name", size: 1}]
+    if miq_request.options[:volumes]
+      phase_context[:requested_volumes] = create_requested_volumes(miq_request.options[:volumes])
+      signal :poll_volumes_complete
+    else
+      signal :prepare_provision
+    end
+  end
+
+  def poll_volumes_complete
+    status, status_message = do_volume_creation_check(phase_context[:requested_volumes])
+    status_message = "completed prepare provision work queued" if status
+    message = "Volume creation is #{status_message}"
```

Comment:
In that case it will log "Volume creation is pending" ("pending" status is returned from fog).